community/kernel/src/test/java/org/neo4j/kernel/impl/transaction/state/NeoStoreFileListingTest.java

```diff
@@ -20,238 +20,120 @@
+        assertFalse( Collections.disjoint( actual, countStoreFiles ) );
     }
 
     @Test
-    public void shouldListLabelScanStoreAndSchemaIndexes() throws Exception
+    public void shouldIncludeTransactionLogFile() throws Exception
     {
-        // Given
-        filesInStoreDirAre( STANDARD_STORE_DIR_FILES, STANDARD_STORE_DIR_DIRECTORIES );
-        scanStoreFilesAre( new String[]{"blah/scan.store", "scan.more"} );
-        indexFilesAre( new String[]{"schema/index/my.index", "schema/index/their.index"} );
-        NeoStoreFileListing fileListing = newFileListing();
-
-        // When
-        ResourceIterator<File> result = fileListing.listStoreFiles( false );
-
-        // Then
-        assertThat( asSetOfPaths( result ), equalTo( asSet(
-                "blah/scan.store",
-                "scan.more",
-                "schema/index/my.index",
-                "schema/index/their.index",
-                "neostore.labeltokenstore.db",
-                "neostore.labeltokenstore.db.names",
-                "neostore.counts.db.a",
-                "neostore.counts.db.b",
-                "neostore.nodestore.db",
-                "neostore.nodestore.db.labels",
-                "neostore.propertystore.db",
-                "neostore.propertystore.db.arrays",
-                "neostore.propertystore.db.index",
-                "neostore.propertystore.db.index.keys",
-                "neostore.propertystore.db.strings",
-                "neostore.relationshipstore.db",
-                "neostore.relationshiptypestore.db",
-                "neostore.relationshiptypestore.db.names",
-                "neostore.schemastore.db",
-                "neostore" ) ) );
+        final ResourceIterator<StoreFileMetadata> storeFiles = neoStoreDataSource.listStoreFiles( true );
+        final Set<String> actual = asSetOfPaths( storeFiles );
+        final Set<String> expectedStoreFiles = expectedStoreFiles( true );
+        final List<String> countStoreFiles = countStoreFiles();
+        assertTrue( actual.containsAll( expectedStoreFiles ) );
+        assertFalse( Collections.disjoint( actual, countStoreFiles ) );
     }
 
     @Test
     public void shouldCloseIndexAndLabelScanSnapshots() throws Exception
     {
         // Given
-        filesInStoreDirAre( STANDARD_STORE_DIR_FILES, STANDARD_STORE_DIR_DIRECTORIES );
-        ResourceIterator<File> scanSnapshot = scanStoreFilesAre( new String[]{"blah/scan.store", "scan.more"} );
-        ResourceIterator<File> indexSnapshot = indexFilesAre( new String[]{"schema/index/my.index"} );
-        NeoStoreFileListing fileListing = newFileListing();
+        LabelScanStore labelScanStore = mock( LabelScanStore.class );
+        IndexingService indexingService = mock( IndexingService.class );
+        LegacyIndexProviderLookup legacyIndexes = mock( LegacyIndexProviderLookup.class );
+        when( legacyIndexes.all() ).thenReturn( Collections.emptyList() );
+        File storeDir = mock( File.class );
```

Comment:
You need this mock to produce some reasonable output on calls to `listFiles()`, as the test will otherwise run into an NPE.